Update kubernetes packages

larvacea opened the pull request:
**Description of changes:**

Updates kubernetes-1.24 through 1.31 to the latest upstream versions. Moves kubernetes-1.27 from standard support to extended.

**Testing done:**

Built AMIs, created nodegroups with eksctl, verified that they would each join the cluster and run busybox. Accidentally verified that an aws-k8s-1.31 node will not join a v1.30 cluster, but will join a v1.31 cluster.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

larvacea commented:
A little more detail on the advisories: these are all vulnerabilities from govulncheck. One is rated High, one Low. The CVE descriptions are a little misleading. Both CVEs talk about `runc` and vulnerable or mitigated `runc` versions. The vulnerabilities and mitigations are in shared code (the `libcontainer` module), and `govulncheck` found that several kubernetes binaries were vulnerable.

larvacea commented:
The severity ratings for the advisories came from NVD (the high-severity CVE) and GitHub (the low-severity CVE). NVD hasn't analyzed the low-severity CVE yet. High severity: container can open arbitrary things on the host. Low severity: container can create new empty files and directories on the host (but can't overwrite existing files).

bcressey commented on packages/kubernetes-1.26/Cargo.toml:
```diff
@@ -69,6 +69,10 @@ sha512 = "6267486606741f25e12b2d0bd19d3243c485b21cce4ae992258a64db714c42ea3542c0
 sha512 = "7b65bb30ee2863ea19dbcb74a855c2a898b85e99e43d772228eec81859017cad0949dd0758461a2c6f7c496907718153ddbd2c30eb578c6b5c11599d593ac72d"
 
+[[package.metadata.build-package.external-files]]
+url = " https://raw.githubusercontent.com/aws/eks-distro/refs/heads/main/projects/kubernetes/kubernetes/1-26/patches/0016-EKS-PATCH-Add-sourceARN-to-sts-headers.patch"
```
nit:
```suggestion
url = "https://raw.githubusercontent.com/aws/eks-distro/refs/heads/main/projects/kubernetes/kubernetes/1-26/patches/0016-EKS-PATCH-Add-sourceARN-to-sts-headers.patch"
```

bcressey commented on packages/kubernetes-1.27/0003-EKS-PATCH-add-Authentication-tracking-request-error-.patch:
```diff
@@ -0,0 +1,307 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
```
No need to commit these patches since they'll be downloaded from the lookaside cache.

bcressey commented on packages/kubernetes-1.27/Cargo.toml:
```diff
@@ -14,8 +14,60 @@ path = "../packages.rs"
+sha512 = "fba641f745b5eef6f17e16501025959e570ae88912d79e2e08b4c79d18475430b879f98947d516871407feb297d5dac9b2fbc4ca565d3770c600d9550f3cffac"
+
+[[package.metadata.build-package.external-files]]
+url = "https://github.com/aws/eks-distro/blob/main/projects/kubernetes/kubernetes/1-27/patches/0001-EKS-PATCH-admission-webhook-exclusion-from-file.patch"
```
Need to use the raw URLs, otherwise the hashes won't match when retrieved via `curl`:
```suggestion
url = "https://raw.githubusercontent.com/aws/eks-distro/refs/heads/main/projects/kubernetes/kubernetes/1-27/patches/0001-EKS-PATCH-admission-webhook-exclusion-from-file.patch"
```

larvacea committed:
packages: update kubernetes 1.31

Update to v1-31-eks-4.

larvacea committed:
packages: update kubernetes-1.30

Update to upstream v1-30-eks-15

larvacea committed:
packages: update kubernetes-1.29

Upstream v1-29-eks-22
k

larvacea committed:
packages: update kubernetes-1.28

Update to upstream v1-28-eks-33

larvacea committed:
packages: update kubernetes-1.27

Update to 1.27.16, add .gitignore for patches

larvacea committed:
packages: update kubernetes-1.26

Add upstream patch 0016.

larvacea committed:
packages: update kubernetes-1.25

Add patch 0025

larvacea committed:
packages: update kubernetes-1.24

Add upstream patch 0037

larvacea committed:
advisories: kubernetes packages

Six advisories: two vulnerabilites each in three kubernetes packages,
kubernetes-1.27, kubernetes-1.28, and kubernetes-1.29.

larvacea commented:
Updated to address Ben's comments.

bcressey commented on packages/kubernetes-1.27/Cargo.toml:
```diff
@@ -14,8 +14,60 @@ path = "../packages.rs"
+sha512 = "fba641f745b5eef6f17e16501025959e570ae88912d79e2e08b4c79d18475430b879f98947d516871407feb297d5dac9b2fbc4ca565d3770c600d9550f3cffac"
+
+[[package.metadata.build-package.external-files]]
+url = "https://raw.githubusercontent.com/aws/eks-distro/refs/heads/main/projects/kubernetes/kubernetes/1-27/patches/0001-EKS-PATCH-admission-webhook-exclusion-from-file.patch"
```
Please add a `.gitignore` to k8s 1.27 through 1.31 to ignore these patches, like the ones in 1.24 - 1.26.

KCSesh commented:
Your commit for 1.29 has a comma instead of a period. If you could fix that, that would be great!

larvacea committed:
packages: add .gitignore to kubernetes packages

Add .gitignore for upstream patches to kubernetes 1.28 and later,
in anticipation of patch files during extended support.
